source/functions.cpp

```diff
@@ -46,146 +51,69 @@ void print(HCL::variable msg, std::string end/* = \n*/) {
 
 	if (msg.value.size() > 1) output += "}";
 
-	printf("%s%s", output.c_str(), end.c_str());
+	std::printf("%s%s", output.c_str(), end.c_str());
 }
 
 
-int createFolder(std::string path, int mode/* = 0777*/) {
-	int check;
-	std::string fullPath;
-	std::vector<std::string> folders = split(path, "/");
-
-	for (auto folder : folders) {
-		fullPath += folder;
-		check = mkdir(fullPath.c_str(), mode);
-		fullPath += "/";
-	}
-  
-    return check;
+int createFolder(std::string path) {
+    return std::filesystem::create_directories(path);
 }
 
 
 int removeFolder(std::string path) {
-	int check = system(std::string("rm -rf " + path).c_str()); // Fuck windows gidjsflkjdalfksjdafiowolfif
-  
-    return check;
+    return static_cast<int>(std::filesystem::remove_all(path));
 }
 
 
 int createFile(std::string path, std::string content/* = ""*/, bool useUtf8BOM/* = false*/) {
-	int output = 0;
-
-	FILE* f = fopen(path.c_str(), "r");
-	if (f == NULL) {
-		f = fopen(path.c_str(), "w");
-		if (!content.empty() && f != NULL) {
-			if (useUtf8BOM) fprintf(f, "\xEF\xBB\xBF");
-			fprintf(f, "%s", content.c_str());
-		}
-		else if (f == NULL) { output = -1; }
-	}
-	else {
-		output = -1;
-	}
-    fclose(f);
+	std::filesystem::path p = path;
 
-	return output;
+    if (!std::filesystem::create_directories(p.parent_path())) return -1;
+
+    auto out = std::ofstream(p);
+    out << content;
```

Comment:
Please convert these from spaces to tabs, as the project uses tabs for indentation.

Reply:
Apologies, my editor usually detects it automatically but not this time, oversight by myself